addons/api/fnc_setupRadios.sqf

```diff
@@ -0,0 +1,84 @@
+#include "script_component.hpp"
+/*
+ * Author: mrschick
+ * When run locally it will set the currently carried radios to the selected channel numbers.
+ *
+ * Arguments:
+ * [RadioSetting1, RadioSetting2, ...] <ARRAY>
+ * 
+ * RadioSetting: [RadioType, Channel] or
+ *               [RadioType, [Channel, Block]] or
+ *               [RadioType, MHz] or
+ *               [RadioType, [MHz, KHz]]
+ *   0: Radio Base Class <STRING>
+ *   1: Radio Channel/Block/Frequency <INTEGER> or <ARRAY>
+ *
+ * Return Value:
+ * Successful <BOOLEAN>
+ *
+ * Example:
+ * _success = [ ["ACRE_PRC343", [2,3]], ["ACRE_PRC152", 3], ["ACRE_PRC77", [2,4]] ] call acre_api_fnc_setupRadios;
+ * // Will set PRC343 to Ch2 Blk3, PRC152 to Ch3 and PRC77 to 31.15 MHz
+ * 
+ * _success = [ ["ACRE_PRC343", 6], ["ACRE_PRC152", 2], ["ACRE_PRC152", 3] ] call acre_api_fnc_setupRadios;
+ * // Will set PRC343 to Ch6 Blk1, the first PRC152 to Ch2 and the second PRC152 to Ch3
+ *
+ * Public: Yes
+ */
+
+private _settings = _this;
+
+if (count(_settings) isEqualTo 0 || !((_settings select 0) isEqualType [])) exitWith {}; // Abort if argument is empty or not an array
```

Comment:
probably want this reversed and short-circuited?

Reply:
I presumed `count` to be the slightly less expensive check, but you're right, all things considered it makes more sense to run the second broader check first.

Reply:
The only issue is that `!((_settings select 0) isEqualType [])` with `_settings=[]` is skipped due to being invalid code, leading to the exit condition not being triggered.